### ISSUE TITLE
Fix slow queries caused by collections

### ISSUE THREAD

**Issue by bqbn** [mozilla/addons-server#8538]
### Describe the problem and steps to reproduce it:
We see slow queries in database that look like these:

```
# Time: 2018-06-15T17:25:24.142424Z
# User@Host: addonsro[addonsro] @ [172.31.34.108] Id: 5734987
# Query_time: 5.328102 Lock_time: 0.000149 Rows_sent: 20 Rows_examined: 2326408
use addons_mozilla_org;
SET timestamp=1529083524;
SELECT (IFNULL(translations.`localized_string`, T3.`localized_string`)) AS `translated_name`, `collections`.`id`, `collections`.`created`, `collections`.`modified`, `collections`.`uuid`, `collections`.`name`, `collections`.`nickname`, `collections`.`slug`, `collections`.`description`, `collections`.`defaultlocale`, `collections`.`collection_type`, `collections`.`icontype`, `collections`.`listed`, `collections`.`subscribers`, `collections`.`downloads`, `collections`.`weekly_subscribers`, `collections`.`monthly_subscribers`, `collections`.`application_id`, `collections`.`addonCount`, `collections`.`upvotes`, `collections`.`downvotes`, `collections`.`rating`, `collections`.`all_personas`, `collections`.`author_id` FROM `collections` LEFT OUTER JOIN `translations` ON (`collections`.`name` = `translations`.`id` AND `translations`.`locale` = 'pt-br') LEFT OUTER JOIN `translations` T3 ON (`collections`.`name` = T3.`id` AND T3.`locale` = `collections`.`defaultlocale`) WHERE (('pt-br'='pt-br') AND `collections`.`listed` = 1 AND (`collections`.`application_id` = 1 OR `collections`.`application_id` IS NULL) AND NOT (`collections`.`addonCount` = 0) AND ((translations.`localized_string` IS NOT NULL OR T3.`localized_string` IS NOT NULL))) ORDER BY `translated_name` ASC LIMIT 20;

# Time: 2018-06-15T17:26:04.811136Z
# User@Host: addonsrw[addonsrw] @ [172.31.5.218] Id: 10507564
# Query_time: 5.112743 Lock_time: 0.000202 Rows_sent: 3 Rows_examined: 2175980
use addons_mozilla_org;
SET timestamp=1529083564;
SELECT DISTINCT (
select 1 from addons_collections as ac
where ac.addon_id = 552696 and ac.collection_id = collections.id
limit 1) AS `has_addon`, `collections`.`id`, `collections`.`created`, `collections`.`modified`, `collections`.`uuid`, `collections`.`name`, `collections`.`nickname`, `collections`.`slug`, `collections`.`description`, `collections`.`defaultlocale`, `collections`.`collection_type`, `collections`.`icontype`, `collections`.`listed`, `collections`.`subscribers`, `collections`.`downloads`, `collections`.`weekly_subscribers`, `collections`.`monthly_subscribers`, `collections`.`application_id`, `collections`.`addonCount`, `collections`.`upvotes`, `collections`.`downvotes`, `collections`.`rating`, `collections`.`all_personas`, `collections`.`author_id`, `translations`.`localized_string` FROM `collections` LEFT OUTER JOIN `collections_users` ON ( `collections`.`id` = `collections_users`.`collection_id` ) LEFT OUTER JOIN `translations` ON ( `collections`.`name` = `translations`.`id` ) WHERE (('de'='de') AND (`collections`.`author_id` = 8930868 OR `collections_users`.`user_id` = 8930868)) ORDER BY `translations`.`localized_string` ASC;

# Time: 2018-06-15T17:26:46.894303Z
# User@Host: addonsrw[addonsrw] @ [172.31.30.38] Id: 10507158
# Query_time: 5.035207 Lock_time: 0.000174 Rows_sent: 3 Rows_examined: 2175980
SET timestamp=1529083606;
SELECT DISTINCT (
select 1 from addons_collections as ac
where ac.addon_id = 371918 and ac.collection_id = collections.id
limit 1) AS `has_addon`, `collections`.`id`, `collections`.`created`, `collections`.`modified`, `collections`.`uuid`, `collections`.`name`, `collections`.`nickname`, `collections`.`slug`, `collections`.`description`, `collections`.`defaultlocale`, `collections`.`collection_type`, `collections`.`icontype`, `collections`.`listed`, `collections`.`subscribers`, `collections`.`downloads`, `collections`.`weekly_subscribers`, `collections`.`monthly_subscribers`, `collections`.`application_id`, `collections`.`addonCount`, `collections`.`upvotes`, `collections`.`downvotes`, `collections`.`rating`, `collections`.`all_personas`, `collections`.`author_id`, `translations`.`localized_string` FROM `collections` LEFT OUTER JOIN `collections_users` ON ( `collections`.`id` = `collections_users`.`collection_id` ) LEFT OUTER JOIN `translations` ON ( `collections`.`name` = `translations`.`id` ) WHERE (('de'='de') AND (`collections`.`author_id` = 8930868 OR `collections_users`.`user_id` = 8930868)) ORDER BY `translations`.`localized_string` ASC;

# Time: 2018-06-15T17:26:38.754822Z
# User@Host: addonsro[addonsro] @ [172.31.43.231] Id: 5735029
# Query_time: 5.364472 Lock_time: 0.000134 Rows_sent: 3 Rows_examined: 2175979
SET timestamp=1529083598;
SELECT DISTINCT (
select 1 from addons_collections as ac
where ac.addon_id = 371918 and ac.collection_id = collections.id
limit 1) AS `has_addon`, `collections`.`id`, `collections`.`created`, `collections`.`modified`, `collections`.`uuid`, `collections`.`name`, `collections`.`nickname`, `collections`.`slug`, `collections`.`description`, `collections`.`defaultlocale`, `collections`.`collection_type`, `collections`.`icontype`, `collections`.`listed`, `collections`.`subscribers`, `collections`.`downloads`, `collections`.`weekly_subscribers`, `collections`.`monthly_subscribers`, `collections`.`application_id`, `collections`.`addonCount`, `collections`.`upvotes`, `collections`.`downvotes`, `collections`.`rating`, `collections`.`all_personas`, `collections`.`author_id`, `translations`.`localized_string` FROM `collections` LEFT OUTER JOIN `collections_users` ON ( `collections`.`id` = `collections_users`.`collection_id` ) LEFT OUTER JOIN `translations` ON ( `collections`.`name` = `translations`.`id` ) WHERE (('de'='de') AND (`collections`.`author_id` = 8930868 OR `collections_users`.`user_id` = 8930868)) ORDER BY `translations`.`localized_string` ASC;

# Time: 2018-06-15T17:26:53.838740Z
# User@Host: addonsrw[addonsrw] @ [172.31.5.218] Id: 10507564
# Query_time: 5.031229 Lock_time: 0.000146 Rows_sent: 3 Rows_examined: 2175979
SET timestamp=1529083613;
SELECT DISTINCT (
select 1 from addons_collections as ac
where ac.addon_id = 371918 and ac.collection_id = collections.id
limit 1) AS `has_addon`, `collections`.`id`, `collections`.`created`, `collections`.`modified`, `collections`.`uuid`, `collections`.`name`, `collections`.`nickname`, `collections`.`slug`, `collections`.`description`, `collections`.`defaultlocale`, `collections`.`collection_type`, `collections`.`icontype`, `collections`.`listed`, `collections`.`subscribers`, `collections`.`downloads`, `collections`.`weekly_subscribers`, `collections`.`monthly_subscribers`, `collections`.`application_id`, `collections`.`addonCount`, `collections`.`upvotes`, `collections`.`downvotes`, `collections`.`rating`, `collections`.`all_personas`, `collections`.`author_id`, `translations`.`localized_string` FROM `collections` LEFT OUTER JOIN `collections_users` ON ( `collections`.`id` = `collections_users`.`collection_id` ) LEFT OUTER JOIN `translations` ON ( `collections`.`name` = `translations`.`id` ) WHERE (('de'='de') AND (`collections`.`author_id` = 8930868 OR `collections_users`.`user_id` = 8930868)) ORDER BY `translations`.`localized_string` ASC;

# Time: 2018-06-15T16:32:29.603173Z
# User@Host: addonsro[addonsro] @ [172.31.10.131] Id: 5578229
# Query_time: 6.023977 Lock_time: 0.000124 Rows_sent: 20 Rows_examined: 2439479
use addons_mozilla_org;
SET timestamp=1529080349;
SELECT (IFNULL(translations.`localized_string`, T3.`localized_string`)) AS `translated_name`, `collections`.`id`, `collections`.`created`, `collections`.`modified`, `collections`.`uuid`, `collections`.`name`, `collections`.`nickname`, `collections`.`slug`, `collections`.`description`, `collections`.`defaultlocale`, `collections`.`collection_type`, `collections`.`icontype`, `collections`.`listed`, `collections`.`subscribers`, `collections`.`downloads`, `collections`.`weekly_subscribers`, `collections`.`monthly_subscribers`, `collections`.`application_id`, `collections`.`addonCount`, `collections`.`upvotes`, `collections`.`downvotes`, `collections`.`rating`, `collections`.`all_personas`, `collections`.`author_id` FROM `collections` LEFT OUTER JOIN `translations` ON (`collections`.`name` = `translations`.`id` AND `translations`.`locale` = 'zh-cn') LEFT OUTER JOIN `translations` T3 ON (`collections`.`name` = T3.`id` AND T3.`locale` = `collections`.`defaultlocale`) WHERE (('zh-cn'='zh-cn') AND `collections`.`listed` = 1 AND (`collections`.`application_id` = 1 OR `collections`.`application_id` IS NULL) AND NOT (`collections`.`addonCount` = 0) AND ((translations.`localized_string` IS NOT NULL OR T3.`localized_string` IS NOT NULL))) ORDER BY `translated_name` ASC LIMIT 20 OFFSET 114320;

```

### What happened?
These slow queries happened during a period of time when we see spikes in datadog dashboard, such as [1] and [2]. However, there is no application logs that proves the link between the two.

[1] https://app.datadoghq.com/dash/391675/amo-prod-frontend-apis-usageperformance
[2] https://app.datadoghq.com/dash/831458/amo?live=true&fullscreen=317799673

### What did you expect to happen?

Please help check the above slow queries are expected, and if they're valid, in what frequency should we expect to see them. If they're not valid ones, could you help improve the query time?

### Anything else we should know?
(Please include a link to the page, screenshots and any relevant files.)

**Comment by diox** [comment]
Going to look if we can remove the pages using those queries right now on the legacy site.

**Comment by AlexandraMoga** [comment]
I've checked the removed features on AMO stage:
1. Public collections, except for the Featured collections, are no longer listed [pic.1]
2. Sort options (i.e. Recently added, Most followers, Name etc) have been removed from the sidebar and the header menu [pic.1]
3. Collection feeds have been removed [pic.1]
4. Contributors part from Edit collection has been removed [pic.2]


@diox 
- Users with `Collection:Contribute` are not able to edit the featured personas collection - https://addons.allizom.org/en-US/firefox/collections/mozilla/featured-personas/ - if they belong only to this group. It worked only after I've added the user to `Staff` as well
- I'm not sure if by....
>Collection admin edition (the functionality is kept but instead moved to the django admin)

  ...you refer to https://addons-internal.stage.mozaws.net/en-US/admin/models/bandwagon/collection/

[pic.1]
![image](https://user-images.githubusercontent.com/31961530/41661290-0515ceb6-74a7-11e8-8208-6bd612da253e.png)
[pic.2]
![image](https://user-images.githubusercontent.com/31961530/41661319-19d03dbe-74a7-11e8-96dc-83ce2a601fdc.png)

**Comment by diox** [comment]
Admins used to see an extra admin tab in the collection edit window. That's what I've removed.

`Collection:Contribute` not working alone is a concern. Can you file a new issue for this? It'll be easier to track down, and I think we can live with this for a week.

**Comment by AlexandraMoga** [comment]
>Admins used to see an extra admin tab in the collection edit window. That's what I've removed.

I understand now. Admin users don't have this extra tab any longer when editing a collection.

Filed #8596 for the permission issue.

Marking this issue as verified fixed.
